tools/cli/src/cli/common.rs

```diff
@@ -158,22 +176,81 @@ impl CommonOpt {
                                 "Privileges have expired for {} - you need to re-authenticate again.",
                                 uat.spn
                             );
-                            std::process::exit(1);
+                            return Err(ToClientError::NeedReauth(username));
                         }
                     }
                 }
             }
             Err(e) => {
                 error!("Unable to read token for requested user - you may need to login again.");
                 debug!(?e, "JWT Error");
-                std::process::exit(1);
+                return Err(ToClientError::NeedLogin(username));
             }
         };
 
         // Set it into the client
         client.set_token(token).await;
 
-        client
+        Ok(client)
+    }
+
+    #[async_recursion]
+    pub async fn to_client(&self, optype: OpType) -> KanidmClient {
+        match self.try_to_client(optype.clone()).await {
+            Ok(c) => c,
+            Err(e) => {
+                match e {
+                    ToClientError::NeedLogin(username) => {
+                        let login_opt = Select::with_theme(&ColorfulTheme::default())
```

Comment:
This should just be a [dialoguer::Confirm](https://docs.rs/dialoguer/0.10.4/dialoguer/struct.Confirm.html)